app/views/groups/group_links/_namespace_group_link.html.erb

```diff
@@ -32,12 +32,66 @@
     </div>
   </td>
   <td class="px-6 py-4">
-    <%= t(
-      :"groups.group_links.index.access_level.level_#{namespace_group_link.group_access_level}"
-    ) %>
+    <div class="form-field">
+      <% if allowed_to?(:update_namespace_with_group_link?, @namespace) && !member_source.key?(:inherited_namespace_path) %>
+        <%= form_with(model: namespace_group_link, url: group_group_link_path(@namespace, namespace_group_link), method: :patch) do |form| %>
+          <%= form.select(
+            :group_access_level,
+            @access_levels,
+            { selected: namespace_group_link.group_access_level },
+            {
+              id: "invited-group-#{namespace_group_link.group.id}-access-level-select",
+              onchange: "this.form.requestSubmit();",
+              "aria-label": t(:"groups.group_links.index.aria_labels.group_access_level")
+            }
+          ) %>
+          <input type="hidden" name="format" value="turbo_stream"/>
+        <% end %>
+      <% else %>
+        <%= t(
+          :"groups.group_links.index.access_level.level_#{namespace_group_link.group_access_level}"
+        ) %>
+      <% end %>
+    </div>
   </td>
   <td class="px-6 py-4">
-    <%= namespace_group_link.expires_at %>
+    <% if allowed_to?(:update_namespace_with_group_link?, @namespace) && !member_source.key?(:inherited_namespace_path) %>
+      <%= form_with( model: namespace_group_link, url: group_group_link_path(@namespace, namespace_group_link), method: :patch) do |form| %>
+        <div class="form-field datepicker" data-controller="datepicker">
+          <div class="relative max-w-sm">
+            <div
+              class="
+                absolute
+                inset-y-0
+                left-0
+                flex
+                items-center
+                pl-3
+                pointer-events-none
+              "
+            >
+              <%= render Viral::IconComponent.new(
```

Comment:
**nitpick:** Sorry, missed this.  Remember this can be rendered as `viral_icon`

Reply:
Updated in a8a148b769c55d47ab10f4413066939b292ca914